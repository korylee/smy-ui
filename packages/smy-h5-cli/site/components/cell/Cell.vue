<template>
  <div v-on="$listeners" class="smy-site-cell">
    <div v-if="$slots.icon" class="smy-site-cell__icon">
      <slot name="icon" />
    </div>
    <div class="smy-site-cell__content">
      <div class="smy-site-cell__title">
        <slot>{{ title }}</slot>
      </div>
      <div v-if="$slots.desc || desc" class="smy-site-cell_desc">
        <slot name="desc">{{ desc }}</slot>
      </div>
    </div>
    <div v-if="$slots.extra" class="smy-site-cell__extra">
      <slot name="extra"> </slot>
    </div>
  </div>
</template>

<script>
import { props } from './props'

export default {
  name: 'SmySiteCell',
  props,
}
</script>

<style lang="less">
@import './cell.less';
</style>
