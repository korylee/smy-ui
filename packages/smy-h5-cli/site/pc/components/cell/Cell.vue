<template>
  <div class="smy-site-cell">
    <div class="var-site-cell__content">
      <div class="var-site-cell__title" :class="titleClass">
        <slot>{{ title }}</slot>
      </div>
      <div class="var-site-cell__desc" :class="descClass" v-if="hasSlots('desc') || desc">
        <slot name="desc">
          {{ desc }}
        </slot>
      </div>
    </div>
  </div>
</template>

<script>
import { props } from './props'
export default {
  name: 'SmySiteCell',
  components: {},
  props,
  methods: {
    hasSlots(name = 'default') {
      return this.$scopedSlots[name] || this.$slots[name]
    },
  },
}
</script>

<style lang="less">
@import './cell.less';
</style>
