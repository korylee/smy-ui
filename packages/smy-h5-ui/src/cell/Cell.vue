<template>
  <div :class="classes" v-on="$listeners" class="smy-cell">
    <div v-if="$slots.icon" class="smy-cell__icon">
      <slot name="icon" />
    </div>
    <div class="smy-cell__content">
      <div class="smy-cell__title" :class="[titleClass ? titleClass : null]">
        <slot>{{ title }}</slot>
      </div>
      <div class="smy-cell__desc" :class="[descClass ? descClass : null]" v-if="$slots.desc || desc">
        <slot name="desc">
          {{ desc }}
        </slot>
      </div>
    </div>
    <div v-if="$slots.extra" class="smy-cell__extra">
      <slot name="extra" />
    </div>
  </div>
</template>

<script>
import { props } from './props'

export default {
  name: 'SmyCell',
  props,
  computed: {
    classes({ border, clickable, insert }) {
      return {
        'smy-cell--border': border,
        'smy-cell--insert': insert,
        'smy-cell--clickable': clickable,
      }
    },
  },
}
</script>

<style lang="less">
@import './cell.less';
</style>
