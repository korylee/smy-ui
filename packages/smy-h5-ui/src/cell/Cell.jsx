import { createNamespace } from '../_utils/vue/create'
import { props } from './props'

import './cell.less'
import { defineComponent, h } from 'vue'

const [name, bem] = createNamespace('cell')

export default defineComponent({
  name,
  props,
  setup(props, { slots, listeners }) {
    return () => {
      const renderIcon = () => {
        return slots.icon ? h('div', { class: bem('icon') }, [slots.icon()], 2) : null
      }
      const renderTitle = () => {
        const descVNode = slots.desc?.() ?? props.desc
        return (
          <div class={[bem('title'), props.titleClass]}>
            {(slots.title ?? slots.default)?.() ?? props.title}
            {descVNode ? <div class={[bem('desc'), props.descClass]}>{descVNode}</div> : null}
          </div>
        )
      }
      const renderValue = () => {
        const valueVNode = slots.value?.() || props.value
        return valueVNode ? <div class={bem('value')}>{valueVNode}</div> : null
      }
      const renderExtra = () => {
        return slots.extra ? <div class={[bem('extra'), props.extraClass]}>{slots.extra()}</div> : null
      }

      const data = {
        class: bem({
          borderless: !props.border,
          insert: props.insert,
          clickable: props.clickable,
        }),
        on: listeners,
      }
      return h('div', data, [renderIcon(), renderTitle(), renderValue(), renderExtra()])
    }
  },
})
