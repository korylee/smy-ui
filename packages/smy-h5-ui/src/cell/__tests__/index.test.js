import { mount } from '@vue/test-utils'
import Cell from '..'
import Vue from 'vue'
import example from '../example'

test('test cell example', () => {
  const wrapper = mount(example)
  expect(wrapper.html()).toMatchSnapshot()
  wrapper.destroy()
})

test('test cell plugin', () => {
  Vue.use(Cell)
  expect(Vue.component(Cell.name)).toBeTruthy()
})

test('test cell props', async () => {
  const wrapper = mount(Cell, {
    propsData: {
      title: 'This is Cell',
      desc: 'This is desc',
      border: true,
      iconClass: 'test-icon-class',
      titleClass: 'test-title-class',
      descClass: 'test-desc-class',
      extraClass: 'test-extra-class',
    },
    scopedSlots: {
      extra: '<div>text</div>',
    },
  })
  expect(wrapper.classes('smy-cell--border')).toBe(true)
  expect(wrapper.find('.smy-cell__title').text()).toBe('This is Cell')
  expect(wrapper.find('.smy-cell__desc').text()).toBe('This is desc')
  expect(wrapper.find('.test-extra-class').text()).toBe('text')
  expect(wrapper.html()).toMatchSnapshot()
  wrapper.destroy()
})
