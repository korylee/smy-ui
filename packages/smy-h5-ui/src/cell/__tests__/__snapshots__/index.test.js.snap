// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test cell example 1`] = `
"<div class=\\"cell-example\\">
  <div>基础使用</div>
  <div class=\\"smy-cell\\">
    <!---->
    <div class=\\"smy-cell__content\\">
      <div class=\\"smy-cell__title\\">这里是cell组件</div>
      <!---->
    </div>
    <!---->
  </div>
  <div>展示描述</div>
  <div class=\\"smy-cell\\">
    <!---->
    <div class=\\"smy-cell__content\\">
      <div class=\\"smy-cell__title\\">这里是cell组件
      </div>
      <div class=\\"smy-cell__desc\\">这里是描述</div>
    </div>
    <!---->
  </div>
  <div>展示图标</div>
  <div class=\\"smy-cell\\">
    <div class=\\"smy-cell__icon\\"><span class=\\"smy-icon\\" style=\\"transition: transform 0ms;\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" xmlns:xlink=\\"http://www.w3.org/1999/xlink\\" version=\\"1.1\\" width=\\"24\\" height=\\"24\\" viewBox=\\"0 0 24 24\\"><path d=\\"M13.46,12L19,17.54V19H17.54L12,13.46L6.46,19H5V17.54L10.54,12L5,6.46V5H6.46L12,10.54L17.54,5H19V6.46L13.46,12Z\\"></path></svg></span></div>
    <div class=\\"smy-cell__content\\">
      <div class=\\"smy-cell__title\\">这里是cell组件
      </div>
      <div class=\\"smy-cell__desc\\">这里是描述</div>
    </div>
    <!---->
  </div>
  <div>右侧内容</div>
  <div class=\\"smy-cell\\">
    <!---->
    <div class=\\"smy-cell__content\\">
      <div class=\\"smy-cell__title\\">这里是cell组件
      </div>
      <!---->
    </div>
    <div class=\\"smy-cell__extra\\"><span class=\\"smy-icon\\" style=\\"transition: transform 0ms;\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" xmlns:xlink=\\"http://www.w3.org/1999/xlink\\" version=\\"1.1\\" width=\\"24\\" height=\\"24\\" viewBox=\\"0 0 24 24\\"><path d=\\"M13.46,12L19,17.54V19H17.54L12,13.46L6.46,19H5V17.54L10.54,12L5,6.46V5H6.46L12,10.54L17.54,5H19V6.46L13.46,12Z\\"></path></svg></span></div>
  </div>
  <div>点击动画</div>
  <div class=\\"smy-cell smy-cell--clickable\\">
    <!---->
    <div class=\\"smy-cell__content\\">
      <div class=\\"smy-cell__title\\">这里是cell组件</div>
      <!---->
    </div>
    <!---->
  </div>
  <div>卡片样式</div>
  <div class=\\"smy-cell insert-example smy-cell--insert\\">
    <!---->
    <div class=\\"smy-cell__content\\">
      <div class=\\"smy-cell__title\\">这里是cell组件
      </div>
      <!---->
    </div>
    <div class=\\"smy-cell__extra\\"><span class=\\"smy-icon\\" style=\\"transition: transform 0ms;\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" xmlns:xlink=\\"http://www.w3.org/1999/xlink\\" version=\\"1.1\\" width=\\"24\\" height=\\"24\\" viewBox=\\"0 0 24 24\\"><path d=\\"M13.46,12L19,17.54V19H17.54L12,13.46L6.46,19H5V17.54L10.54,12L5,6.46V5H6.46L12,10.54L17.54,5H19V6.46L13.46,12Z\\"></path></svg></span></div>
  </div>
  <div>分组用法</div>
  <div class=\\"smy-cell-group\\">
    <div class=\\"smy-cell smy-cell--border smy-cell--clickable\\">
      <!---->
      <div class=\\"smy-cell__content\\">
        <div class=\\"smy-cell__title\\">这里是cell组件
        </div>
        <div class=\\"smy-cell__desc\\">这里是描述</div>
      </div>
      <div class=\\"smy-cell__extra\\"><span class=\\"smy-icon\\" style=\\"transition: transform 0ms;\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" xmlns:xlink=\\"http://www.w3.org/1999/xlink\\" version=\\"1.1\\" width=\\"24\\" height=\\"24\\" viewBox=\\"0 0 24 24\\"><path d=\\"M13.46,12L19,17.54V19H17.54L12,13.46L6.46,19H5V17.54L10.54,12L5,6.46V5H6.46L12,10.54L17.54,5H19V6.46L13.46,12Z\\"></path></svg></span></div>
    </div>
    <div class=\\"smy-cell smy-cell--border smy-cell--clickable\\">
      <!---->
      <div class=\\"smy-cell__content\\">
        <div class=\\"smy-cell__title\\">这里是cell组件
        </div>
        <div class=\\"smy-cell__desc\\">这里是描述</div>
      </div>
      <div class=\\"smy-cell__extra\\"><span class=\\"smy-icon\\" style=\\"transition: transform 0ms;\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" xmlns:xlink=\\"http://www.w3.org/1999/xlink\\" version=\\"1.1\\" width=\\"24\\" height=\\"24\\" viewBox=\\"0 0 24 24\\"><path d=\\"M13.46,12L19,17.54V19H17.54L12,13.46L6.46,19H5V17.54L10.54,12L5,6.46V5H6.46L12,10.54L17.54,5H19V6.46L13.46,12Z\\"></path></svg></span></div>
    </div>
    <div class=\\"smy-cell smy-cell--border smy-cell--clickable\\">
      <!---->
      <div class=\\"smy-cell__content\\">
        <div class=\\"smy-cell__title\\">这里是cell组件
        </div>
        <div class=\\"smy-cell__desc\\">这里是描述</div>
      </div>
      <div class=\\"smy-cell__extra\\"><span class=\\"smy-icon\\" style=\\"transition: transform 0ms;\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" xmlns:xlink=\\"http://www.w3.org/1999/xlink\\" version=\\"1.1\\" width=\\"24\\" height=\\"24\\" viewBox=\\"0 0 24 24\\"><path d=\\"M13.46,12L19,17.54V19H17.54L12,13.46L6.46,19H5V17.54L10.54,12L5,6.46V5H6.46L12,10.54L17.54,5H19V6.46L13.46,12Z\\"></path></svg></span></div>
    </div>
  </div>
  <div>分组卡片样式</div>
  <div class=\\"smy-cell-group insert-example smy-cell-group--insert\\">
    <div class=\\"smy-cell smy-cell--border smy-cell--clickable\\">
      <!---->
      <div class=\\"smy-cell__content\\">
        <div class=\\"smy-cell__title\\">这里是cell组件
        </div>
        <div class=\\"smy-cell__desc\\">这里是描述</div>
      </div>
      <div class=\\"smy-cell__extra\\"><span class=\\"smy-icon\\" style=\\"transition: transform 0ms;\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" xmlns:xlink=\\"http://www.w3.org/1999/xlink\\" version=\\"1.1\\" width=\\"24\\" height=\\"24\\" viewBox=\\"0 0 24 24\\"><path d=\\"M13.46,12L19,17.54V19H17.54L12,13.46L6.46,19H5V17.54L10.54,12L5,6.46V5H6.46L12,10.54L17.54,5H19V6.46L13.46,12Z\\"></path></svg></span></div>
    </div>
    <div class=\\"smy-cell smy-cell--border smy-cell--clickable\\">
      <!---->
      <div class=\\"smy-cell__content\\">
        <div class=\\"smy-cell__title\\">这里是cell组件
        </div>
        <div class=\\"smy-cell__desc\\">这里是描述</div>
      </div>
      <div class=\\"smy-cell__extra\\"><span class=\\"smy-icon\\" style=\\"transition: transform 0ms;\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" xmlns:xlink=\\"http://www.w3.org/1999/xlink\\" version=\\"1.1\\" width=\\"24\\" height=\\"24\\" viewBox=\\"0 0 24 24\\"><path d=\\"M13.46,12L19,17.54V19H17.54L12,13.46L6.46,19H5V17.54L10.54,12L5,6.46V5H6.46L12,10.54L17.54,5H19V6.46L13.46,12Z\\"></path></svg></span></div>
    </div>
    <div class=\\"smy-cell smy-cell--border smy-cell--clickable\\">
      <!---->
      <div class=\\"smy-cell__content\\">
        <div class=\\"smy-cell__title\\">这里是cell组件
        </div>
        <div class=\\"smy-cell__desc\\">这里是描述</div>
      </div>
      <div class=\\"smy-cell__extra\\"><span class=\\"smy-icon\\" style=\\"transition: transform 0ms;\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" xmlns:xlink=\\"http://www.w3.org/1999/xlink\\" version=\\"1.1\\" width=\\"24\\" height=\\"24\\" viewBox=\\"0 0 24 24\\"><path d=\\"M13.46,12L19,17.54V19H17.54L12,13.46L6.46,19H5V17.54L10.54,12L5,6.46V5H6.46L12,10.54L17.54,5H19V6.46L13.46,12Z\\"></path></svg></span></div>
    </div>
  </div>
</div>"
`;

exports[`test cell props 1`] = `
"<div class=\\"smy-cell smy-cell--border\\" iconclass=\\"test-icon-class\\">
  <!---->
  <div class=\\"smy-cell__content\\">
    <div class=\\"smy-cell__title test-title-class\\">This is Cell</div>
    <div class=\\"smy-cell__desc test-desc-class\\">
      This is desc
    </div>
  </div>
  <div class=\\"smy-cell__extra test-extra-class\\">
    <div>text</div>
  </div>
</div>"
`;
